search-commons/src/test/java/no/unit/nva/search2/ResourceQueryTest.java

```diff
@@ -0,0 +1,84 @@
+package no.unit.nva.search2;
+
+import java.util.stream.Collectors;
+import nva.commons.apigateway.exceptions.BadRequestException;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.net.URI;
+import java.util.stream.Stream;
+
+import static no.unit.nva.search2.ResourceParameter.OFFSET;
+import static no.unit.nva.search2.ResourceParameter.PAGE;
+import static no.unit.nva.search2.ResourceParameter.PER_PAGE;
+import static no.unit.nva.search2.common.OpenSearchQuery.queryToMap;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class ResourceQueryTest {
+
+
+    @ParameterizedTest
+    @MethodSource("uriProvider")
+    void buildUriFromValidUri(URI uri) throws BadRequestException {
```

Comment:
so you are hiding the internal uri from the client if i understand correctly?

Reply:
yes, I do a translation from q params to lucene query